src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx

```diff
@@ -96,82 +118,84 @@ export default class Summary extends React.Component {
+      if (type === 'USER_FAILURE') {
+        // user failure
+        if (runtimeOutput) {
+          const userCode = runtimeOutput.originalUserExitCode;
+          if (userCode) {
+            result.push(<HintItem key='user-exit-code' header='User Exit Code:' value={userCode} />);
           }
         }
-
-        return (
-          <MessageBar messageBarType={MessageBarType.error}>
-            <div>
-              <div>
-                <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                  Error Type:
-                </span>
-                <span>User Error</span>
-              </div>
-              {containerId && (
-                <div>
-                  <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                    Container ID:
-                  </span>
-                  <span>{containerId}</span>
-                </div>
-              )}
-              {userExitCode && (
-                <div>
-                  <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                    Exit Code:
-                  </span>
-                  <span>{userExitCode}</span>
-                </div>
-              )}
-              <div>
-                <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                  Resolution:
-                </span>
-                <span>{`Please check container's Stdout and Stderr for more information.`}</span>
-              </div>
-            </div>
-          </MessageBar>
-        );
       } else {
-        return (
-          <MessageBar messageBarType={MessageBarType.error}>
-            <div>
-              <div>
-                <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                  Error Type:
-                </span>
-                <span>System Error</span>
-              </div>
-              <div>
-                <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
-                  Resolution:
-                </span>
-                <span>Please send the <Link onClick={this.showApplicationSummary}>application summary</Link> to your administrator for further investigation.</span>
-              </div>
-            </div>
-          </MessageBar>
-        );
+        // platform failure
+        const code = jobInfo.jobStatus.appExitCode;
+        result.push(<HintItem key='platform-exit-code' header='Platform Exit Code:' value={code} />);
       }
+      // reason & solution
+      if (runtimeOutput) {
+        const reason = runtimeOutput.reason;
+        const solution = runtimeOutput.solution;
+        if (reason) {
+          result.push(<HintItem key='reason' header='Reason:' value={reason} />);
+        }
+        if (solution) {
+          result.push(<HintItem key='solution' header='Solution:' value={solution} />);
+        }
```

Comment:
We should always show reason and solution from spec, and concate the dynamic reason and solution from runtime if they are available.